doc/neps/nep-0029-deprecation_policy.rst

```diff
@@ -0,0 +1,190 @@
+development guidelines:
+
+
+   - support minor versions of ``Python`` initially released
+     42 months prior to our planned release date
+   - always support at least the 2 latest minor versions of ``Python``
+   - support minor versions of ``numpy`` initially released in the 24
+     months prior to our planned release date or oldest that supports the
+     minimum Python version (whichever is higher)
+
+   The minimum supported version of ``Python`` will be set to
+   ``python_requires`` in ``setup`` and all supported versions of
+   Python will be in the test matrix and have binary artifacts built
+   for releases.
+
+   We will bump the minimum Python and NumPy versions as we can on
+   every minor and major release, but never on a patch release.
+
+For other dependencies, adopt similar time windows of 24 months or
+shorter.
+
+
+Backward compatibility
+----------------------
+
+No issues other than the intentional dropping of old version of
+``Python`` and dependencies.
+
+
+Alternatives
+------------
+
+Ad-Hoc
+~~~~~~
+
+Projects could on every release evaluate if they want to increase
+the minimum version of Python supported.  While this is a notionally
+simple policy, it makes it hard for downstream users to predict what
+the future minimum versions will be.  As there is no objective threshold
+to when the minimum version should be dropped, it is easy for these
+discussions to devolve into [bike shedding](https://en.wikipedia.org/wiki/Wikipedia:Avoid_Parkinson%27s_bicycle-shed_effect) and acrimony.
+
+
+All Python Software Foundation supported versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is a very clear and conservative approach.  However, it means that
+there is 4 year lag between when new language features come into the
+language and when the projects are able to use them.  Additionally,
+for projects that have a significant component of compiled extensions
+this requires building many binary artifacts for each release.
+
+For the case of NumPy, many projects carry workarounds to bugs that
+are fixed in subsequent versions of NumPy.  Being proactive about
+increasing the minimum version of NumPy will allow downstream
+packages to carry fewer version-specific patches.
+
+
+
+Default version on Linux distribution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The policy could be to support the version of Python that ships by
+default in the latest Ubuntu LTS or CentOS/RHEL release.  However, we
+would still have to standardize across the community which
+distribution we are following.
+
+By following the versions supported by major Linux distributions, we
+are giving up technical control of our projects to external
+organizations that may have different motivations and concerns than we
+do.
+
+N minor versions of Python
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the current release cadence of the Python, the proposed time
+(42 months) is roughly equivalent to "the last two" Python minor
```

Comment:
minor and major is a bit confusing when it comes to Python. Perhaps define up front minor = x.x versions and dot releases (x.x.x) and major = x

Reply:
Added a section at the top defining these terms.